design/KEPS/graph-cache/proposal.md

```diff
@@ -0,0 +1,236 @@
+
+The recommended approach is **Background Refresh (Re-compute)**, which provides automatic, permission-aware caching without configuration burden.
+
+# Technical Challenges
+
+One of Kiali's strengths is that it offers the end user a vast number of configuration options, such that views can be customized as needed to help visualize or troubleshoot the mesh. This is a big problem for pre-computing information. The processor and memory resources required to pre-compute the potential combinations is prohibitive.
+
+Note that the focus is on the cross-cluster/cross-namespace pages. When a user has navigated the view to a detail page, on-demand processing is typically both feasible, due to a narrow scope, and desired, even mildly stale data is less acceptable.
+
+## Graph Options
+
+The Traffic Graph offers many options and the option settings directly affect the back-end processing. The namespaces and graph-type alone will fundamentally determine the queries performed and the resulting graph.
+
+This is a difficult situation for pre-compute, as it's not possible to predict what the desired graph will be.
+
+Possible options:
+
+- **Predefinition (pre-compute)** - ❌ **ELIMINATED after analysis (Nov 2025)**
+
+  - Let users define graphs in the Kiali CR. These "favorite" graphs could be pre-computed and rendered quickly.
+  - Benefits:
+    - favorite graphs always ready
+  - Downsides:
+    - it may be very unlikely for users to want to define graph queries in config
+    - config-based queries are fixed may become dated
+    - updates require kiali pod restarts
+    - some users may not have permissions to view the data resulting from the predefined graph
+    - **CRITICAL**: Configuration barrier makes this impractical for real-world usage
+    - **CRITICAL**: Permission mismatch between controller SA and users has no good solution
+
+- **Background refresh (re-compute)** - ✅ **RECOMMENDED (Nov 2025)**
+
+  - Like today, users can set graph options as needed, and then launch a graph request. The user will wait for the initial graph to render. The result will be cached and the server will proactively re-compute the same graph, updating the cache. Requests for the same graph will return the most recent cached result. A change of graph, a cache-hit timeout, or a session end would end the re-compute.
+  - Benefits:
+    - avoids permission issues and config, works with any graph at any time
+    - no user configuration required (automatic, self-service)
+    - permission-aware caching (per-session)
+    - self-optimizing (popular graphs stay cached)
+  - Downsides:
+    - still have to wait on the first request and any subsequent change to the requested graph
+
+# Proposed Implementation
+
+## Architecture
+
+The graph cache implementation uses a **per-session** caching strategy with background refresh jobs. Unlike the controller-based approach used for Validations and Health, the graph cache uses a custom in-memory cache for several key reasons:
```

Comment:
nit: Validations also use a custom in-memory cache. That's basically all `KialiCache` is.